core/src/main/scala/epollcat/internal/ch/socket.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2022 Arman Bilge
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package epollcat.internal.ch
+
+import scala.annotation.nowarn
+import scala.scalanative.unsafe._
+
+@extern
+@nowarn
+private[ch] object socket {
+  final val SOCK_NONBLOCK = 2048
```

Comment:
Perhaps one line to state which operating systems use
this magic value, Linux, macOS, FreeBSD? 

Somebody is going to port to an OS which does not.

Reply:
Yes, even I will forget and make a mistake.

I noted that `SOCK_NONBLOCK` and also `accept4` are supported on Linux and FreeBSD, however we are currently only using them on Linux. Because we do not have a FreeBSD environment in CI I hesitant to add special conditions for it.

Reply:
re: FreeBSD  understood & concur.